src/supplemental/http/http_server.c

```diff
@@ -1146,26 +1146,31 @@ nni_http_server_add_handler(nni_http_server *s, nni_http_handler *h)
 		while ((len2 > 0) && (h2->uri[len2 - 1] == '/')) {
 			len2--; // ignore trailing '/'
 		}
-		if (strncmp(h->uri, h2->uri, len > len2 ? len2 : len) != 0) {
-			continue; // prefixes don't match.
+
+		if (len != len2) {
+			continue; // length mismatch
```

Comment:
I am someone concerned by this change.  There is a semantic change on the URI handler, and while I think it is usually a good one, it won't necessarily always be so.

I think we might want a way to get the old matching behavior for h->tree handlers -- maybe another option.  Essentially I think it may be desirable for some use cases (think nanomsg websocket transports) where ownership of a subtree is "exclusive".

Reply:
Please think about whether you can achieve this.  Also the minor updates requested should be addressed -- and maybe copyright update?

Reply:
Yes, I'll look at this. Off the top of my head, shouldn't the longest uri match resolve this naturally? I.e. having a tree handler at "/tree" will handle "/tree/*", and one at "/" will resolve everything else. But since "/tree" handler lies before "/", the resolving is automagik. I'm not enough savvy about these technologies to see the problem this might incur.

Reply:
My point is that someone registering at /tree might actually want to exclude a registration at /tree/subtree

This is really something that should be configurable.  I'm not disagreeing that your current implementation is not a bad default -- but it is a change, and we need a way to give the old behavior.  (In particular I think the ws:// API for websockets may kind of depend on it.)

Reply:
Something like:
`nng_http_handler_set_tree(nng_http_handler *h, int allow_sub_tree)` ?
or:
`nng_http_handler_set_tree_non_exclusive(nng_http_handler *h)` ?

Reply:
Well, set_tree() is an API that already exists so we don't want to change that.

But I could see having nng_http_handler_set_tree_exclusive() as a variant that gives the exclusive access version.

Probably this should be the default for the websocket transport for nanomsg, but not for other consumers of the HTTP framework.

Reply:
I'll give this a go.